### PR TITLE
Simple JS WP build env + prettier & eslint

### PR DESCRIPTION
**Pull request by jblz**
This is a step toward a more robust JS environment for the front end and admin (ES6+ syntax, importing npm packages, etc.).

It leverages the [`wp-scripts` package](https://www.npmjs.com/package/@wordpress/scripts) so we don't need a custom webpack config (yet).

Additionally, this includes prettier and eslint configurations to enable IDE auto-formatters.

Right now, the src/index.js file is only a placeholder for the entrypoint. We'll want to replace it with an actual implementation prior to sourcing the built file in the plugin :)

## To Test

`npm i` to install dev dependencies.

### Development

* `npm run start`
* Make changes to `src/index.js`, add and link other source files (using `import _ from '__'` format, etc.)
* Source changes should automatically generate a new development `build/index.js`

### Release

* `npm run build`
* It should generate a new "production" `build/index.js`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "build": "wp-scripts build",
+    "start": "wp-scripts start",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Parsely/wp-parsely.git"
+  },
+  "author": "",
+  "license": "GPL-2.0-only",
```
Comment (GaryJones) [inline package.json:19]: This doesn't match what the plugins was already licensed under, which is GPL-2.0-or-later, which is what WP core uses. Keeping it to only 2.0 would technically stop someone using it with a plugin that is only GPL-3.0. 
```suggestion
  "license": "GPL-2.0-or-later",
```

**Review by GaryJones — CHANGES_REQUESTED** [review]
Incorrect license SPDX tag spotted. 

As a general question, the existing plugin obviously has a little bit of JS, but is the plan to introduce a lot more?

**Comment by jblz** [comment]
Thanks for taking a look @GaryJones.  Fixed up the license.

> is the plan to introduce a lot more (js)?

I don't think it'll be _a lot_ off the bat, but as I started tackling issues like #158 & #134, I really missed the ability to write modern JS.

In particular, when we extract the widget script from PHP (#158) it's an opportune time to refactor as we'll be pulling a bunch values from PHP `echo`s into variables.

Further, modularizing the front end code enables us to cover it with tests more readily and things like destructuring assignments (particularly in function arguments) are just so nice to be able to use 😄.

I think having a build step in place will enable feature development in the future as well, so I think it's a pretty good time to make the jump.  Happy to hear counter arguments, of course!

**Review by GaryJones — APPROVED** [review]
👍🏻

**Pull request merged**